### ISSUE TITLE
tuning interface

### ISSUE THREAD

**Issue by accosmin** [accosmin-org/libnano#47]
* the main tuner_t interface should parallelize across folds and hyper-parameter trials
* tuner_t should always work on a dense fixed grid
* tuning strategy: exhaustive grid search
* tuning strategy: random search
* tuning strategy: greedy search around the current best N points in their 3x3 neighbourhoods
* tuning strategy: convex quadratic model of the validation loss wrt hyper-parameters
